lib/Backend/Lower.cpp

```diff
@@ -19489,6 +19535,53 @@ void Lowerer::GenerateFastInlineIsIn(IR::Instr * instr)
     instr->InsertAfter(doneLabel);
 }
 
+// Generate a fast path for the "in" operator to use the cache where the key may be a PropertyString or Symbol.
+void Lowerer::GenerateFastObjectIsIn(IR::Instr * instr)
+{
+    if (!instr->GetSrc1()->IsRegOpnd() || !instr->GetSrc2()->IsRegOpnd())
+    {
+        // REVIEW: Is it worth adding logic for this case? Under what circumstances would the operands not be registers?
```

Comment:
i think in case you have a string literal like if('a' in obj) you might end up with a MemRefOpnd or IndirOpnd

Reply:
Okay, thanks. I went ahead and added the logic to move those to registers.

---
In reply to: [176263822](https://github.com/Microsoft/ChakraCore/pull/4846#discussion_r176263822) [](ancestors = 176263822)